ginga/misc/plugins/Thumbs.py

```diff
@@ -365,6 +373,34 @@ def copy_attrs(self, fitsimage):
                                   ['transforms', 'cutlevels',
                                    'rgbmap'])
 
+    def highlight_thumbnail(self, viewer, fitsimage):
+        """Highlight currently active thumbnail. Called by ``focus_cb()``."""
+        image = fitsimage.get_image()
+        name = image.get('name')
+        path = image.get('path')
+        chname = viewer.get_channelName(fitsimage)
+        thumbkey = self.get_thumb_key(chname, name, path)
+        chname = chname.lower()  # for matching with thumbkey
+
+        hlight_bg = self.settings.get('label_bg_color', 'yellow')
+        hlight_fg = self.settings.get('label_font_color', 'black')
+
+        with self.thmblock:
```

Comment:
This looks like it is iterating over the entire set of thumbnails every time the keyboard focus changes from one widget to another.  At our telescope in the course of a night there are hundreds of thumbs present, and people are changing widgets (tabs) constantly.  I think this is not an efficient way to go about this.

Why don't we keep a set of thumbkeys (one set per channel).  In each set are the keys for the currently highlighted thumbs. When this function is called it is called with the new set of keys that should be highlighted (this is figured out elsewhere, according to the highlight scheme).  All this function has to do is take the difference of the sets (in two ways) to figure out which ones to unhighlight and which ones to highlight.

For example:
```python
In [5]: foo = set(['a', 'b', 'c', 'd'])

In [6]: goo = set(['b', 'c', 'd', 'e'])

In [7]: foo - goo
Out[7]: {'a'}

In [8]: goo - foo
Out[8]: {'e'}
```
If `foo` is the old set of highlighted thumbkeys, and `goo` is the new set, then (`foo` - `goo`) are the ones to *un*highlight, and (`goo` - `foo`) are the ones we need to highlight.